### PR TITLE
Added non-exhaustive enums to the CLI

### PR DESCRIPTION
**Pull request by rdelfin**
Currently the CLI allows for all other enum builder options except for non-exhaustive rustified enums. This PR fixes the omission.

Solves #2846

### CODE REVIEW

**Comment by rdelfin** [comment]
Clippy errors look unrelated to my change but feel free to correct me

File: bindgen/lib.rs [hunk 1]
```diff
@@ -87,10 +87,10 @@ pub const DEFAULT_ANON_FIELDS_PREFIX: &str = "__bindgen_anon_";
 const DEFAULT_NON_EXTERN_FNS_SUFFIX: &str = "__extern";
 
 fn file_is_cpp(name_file: &str) -> bool {
-    name_file.ends_with(".hpp") ||
-        name_file.ends_with(".hxx") ||
-        name_file.ends_with(".hh") ||
-        name_file.ends_with(".h++")
+    name_file.ends_with(".hpp")
```
Comment (rdelfin) [inline bindgen/lib.rs:90]: Oh... This must be my rustfmt settings. Strange, I'd have expected my IDE to grab them from the rustfmt.toml at the top

Comment (pvdrz) [reply bindgen/lib.rs]: you need nightly rust to format bindgen's code

**Review by pvdrz — CHANGES_REQUESTED** [review]
Thanks for catching this. Could you document this change in the Changelog? otherwise it looks good to me :)

File: bindgen/lib.rs [hunk 2]
```diff
@@ -237,6 +237,7 @@ impl std::fmt::Display for Formatter {
 /// 2. [`bitfield_enum()`](#method.bitfield_enum)
 /// 3. [`newtype_enum()`](#method.newtype_enum)
 /// 4. [`rustified_enum()`](#method.rustified_enum)
+/// 4. [`rustified_non_exhaustive_enum()`](#method.rustified_non_exhaustive_enum)
```
Comment (kulp) [inline bindgen/lib.rs:240]: I think the Markdown-formatted version will correct "4" to "5" here, but you might care to change this in the raw source also.
```suggestion
/// 5. [`rustified_non_exhaustive_enum()`](#method.rustified_non_exhaustive_enum)
```

**Comment by rdelfin** [comment]
Sorry for the delay! Should be sorted now

**Review by kulp — APPROVED** [review]

**Review by pvdrz — APPROVED** [review]

**Pull request merged**